webcompat/static/js/lib/labels.js

```diff
@@ -4,31 +4,119 @@
 
  var issues = issues || {};
 
-// Read-only Model of all labels in the repo
-issues.AllLabels = Backbone.Model.extend({
-  url: function() {
-    return '/api/issues/labels';
-  },
-  // See also issues.Issue#removeNamespaces
-  removeNamespaces: function(labelsArray) {
-    // Return a copy of labelsArray with the namespaces removed.
-    var namespaceRegex = /(browser|closed|os|status)-/i;
-    var labelsCopy = _.cloneDeep(labelsArray);
-    return _.map(labelsCopy, function(labelObject) {
-      labelObject.name = labelObject.name.replace(namespaceRegex, '');
-      return labelObject;
+/**
+* A LabelList is a list of labels.
+*
+* It takes care of all namespace prefixing and unprefixing, so that
+* the rest of the app doesn't ever need to worry about those details.
+* To initialize, either pass in a list of labels as an array of strings
+* or an array of objects:
+*
+* new issues.LabelList({labels: ['firefox', 'ie', 'chrome']});
+*
+* new issues.LabelList({labels: [{name:'status-worksforme', url:'...',
+*    color:'cccccc'}]});
+*
+* Or a URL to a JSON file describing the labels:
+*
+* new issues.LabelList({url:'/path/to/labels.json'});
+*/
+
+issues.LabelList = Backbone.Model.extend({
```

Comment:
Now that the label model is more than 5 lines (or whatever), can we split this out and add to `/lib/models/labels.js` or `/lib/models/label-list.js`?